### PR TITLE
Update supported Python versions

### PR DESCRIPTION
**Pull request by cdiener**
* [X] fix CI errors with pytest-benchmark
* [X] description of feature/fix
* [X] tests added/passed
* [X] add an entry to the [next release](../release-notes/next-release.md)

Update Python versions consistent with https://endoflife.date/python .

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -16,7 +17,7 @@ deps=
     jsonschema
     osqp~=0.6 ; python_version < "3.10"
     pytest
-    pytest-benchmark
+    pytest-benchmark~=4.0.0
```
Comment (Midnighter) [inline tox.ini]: Same as for optlang.
```suggestion
    pytest-benchmark~=4.0
```

File: tox.ini [hunk 2]
```diff
@@ -30,7 +31,7 @@ deps=
...
```
Comment (Midnighter) [inline tox.ini]:
```suggestion
    pytest-benchmark~=4.0
```

**Review by Midnighter — APPROVED** [review]
Cheers, thank you!

File: tox.ini (hunk 1 above)
Comment (cdiener) [reply tox.ini]: thanks!

**Comment by cdiener** [comment]
Waiting on the last release for swiglpk to get to PyPi which should fix Python 3.11.

**Pull request merged**